### PR TITLE
Pd 170 marketplace topics

### PR DESCRIPTION
**Pull request by thehannahlerman**

### CODE REVIEW

File: app/routes/__marketing/index.tsx [hunk 1]
```diff
@@ -81,7 +81,7 @@ export default function Index() {
               <p className="text-content">
                 The three-step Brainstorming, Analytics, and Review system provides a variety of ways for the community
                 and Web3 organizations to generate engagement, education, content, and peer reviewed analytics about
-                protocols or topics they care about. Each step can be activated on its own or together.
+                protocols or challenges they care about. Each step can be activated on its own or together.
```
Comment (sebruiz) [inline app/routes/__marketing/index.tsx:84]: I think this marketing lingo probably stays the same (not referring to challenges within the app)

File: app/components/AppHeader.tsx [hunk 2]
```diff
@@ -89,7 +89,7 @@ export function AppHeader({ links, userLinks }: AppHeaderProps) {
   ));
 
   return (
-    <Header height={HEADER_HEIGHT} sx={{ borderBottomWidth: "2px", borderColor: "#FAFAFA" }}>
+    <Header height={HEADER_HEIGHT} sx={{ marginBottom: "64px", borderBottomWidth: "2px", borderColor: "#FAFAFA" }}>
```
[the remark this answers is not in the document]
Comment (sebruiz) [reply app/components/AppHeader.tsx:92]: Also, mental note, I like this idea of uniform spacing between the header and body across pages. We need to go remove any margins/padding we have on other pages.

[the remark this answers is not in the document]
Comment (thehannahlerman) [reply app/components/AppHeader.tsx:92]: I can go through and look at that now too

File: app/hooks/useChallenge.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { useQuery } from "@tanstack/react-query";
```
Comment (bcinman) [inline app/hooks/useChallenge.ts:1]: Don't needs hooks for reads on data that will be indexed in prisma, we can just use remix loaders for that.

File: app/hooks/useChallenges.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { useQuery } from "@tanstack/react-query";
+import type { ChallengeWithMarketplace } from "~/domain";
+import { fakeChallenge } from "~/utils/fakes";
+
+type Params = {};
+
+export function useChallenges(params: Params) {
```
Comment (bcinman) [inline app/hooks/useChallenges.ts:7]: Same here, don't need this hook.

File: app/routes/app/brainstorm/$marketplace.c.new.tsx [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
-export default function CreateTopic() {
+export default function CreateChallenge() {
   const marketplace = true;
 
   return (
     <div className="mx-auto max-w-3xl space-y-10">
       <div className="grid grid-cols-1 md:grid-cols-3 gap-5 items-center">
         <Title order={2} weight={600} className="md:col-span-3">
-          {"Create Topic"}
+          {"Create Challenge"}
```
Comment (bcinman) [inline app/routes/app/brainstorm/$marketplace.c.new.tsx:11]: Static text doesn't need to be in `{""}`s. There's a few instances in this file that can be cleaned up.

`{}` is only necessary when you're injecting a dynamic value. e.g: `<Text>{challenge.title}</Text>`

File: app/routes/app/brainstorm/$marketplace.challenges.tsx [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+      </div>
+    </>
+  );
+}
+
+function ProjectWithIcon({ project }: { project: string }) {
+  const iconUrl = PROJECT_ICONS[project];
+
+  return <TextWithIcon text={project} iconUrl={iconUrl ?? null} />;
+}
```
Comment (termaatFlipside) [inline app/routes/app/brainstorm/$marketplace.challenges.tsx:313]: Maybe combine with/use Project badge since it's used elsewhere

**Review by bcinman — CHANGES_REQUESTED** [review]

File: app/routes/app/brainstorm/$marketplace.challenges.tsx [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+        </main>
+      </section>
+    </div>
+  );
+}
+
+function SearchAndFilter() {
+  return (
+    <Form className="space-y-3 p-3 border-[1px] border-solid border-[#EDEDED] rounded-md bg-brand-400 bg-opacity-5">
+      <Input placeholder="Search" name="q" rightSection={<Search16 />} />
```
Comment (termaatFlipside) [inline app/routes/app/brainstorm/$marketplace.challenges.tsx:123]: These inputs probably need radius="md"

File: app/routes/app/brainstorm/$marketplace.challenges.tsx [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+        Apply Filters
+      </Button>
+    </Form>
+  );
+}
+
+function Prerequisites({ challenge }: { challenge: ChallengeWithMarketplace }) {
+  return (
+    <section className="flex flex-col-reverse md:flex-row space-y-reverse space-y-7 md:space-y-0 space-x-0 md:space-x-5">
+      <main className="flex-1">
```
Comment (termaatFlipside) [inline app/routes/app/brainstorm/$marketplace.challenges.tsx:180]: Can simplify `space-y-7 md:space-y-0 ` to `gap-y-7` same goes for x

File: app/routes/app/brainstorm/$marketplace.challenges.tsx [hunk 9]
```diff
@@ -0,0 +1,324 @@
...
+          </div>
+        </div>
+      </main>
+    </section>
+  );
+}
+
+function Rewards({ challenge }: { challenge: ChallengeWithMarketplace }) {
+  return (
+    <section className="flex flex-col-reverse md:flex-row space-y-reverse space-y-7 md:space-y-0 space-x-0 md:space-x-5">
```
Comment (termaatFlipside) [inline app/routes/app/brainstorm/$marketplace.challenges.tsx:232]: and same `gap-y-7` here

File: app/hooks/useChallenge.ts (hunk 3 above)
Comment (thehannahlerman) [reply app/hooks/useChallenge.ts:1]: Sebastian and I were thinking the same thing

File: app/routes/app/brainstorm/$marketplace.c.new.tsx (hunk 5 above)
Comment (thehannahlerman) [reply app/routes/app/brainstorm/$marketplace.c.new.tsx:11]: oops lol didn't even notice this when I edited it. Did a find and replace I think. Will fix

File: app/routes/app/brainstorm/$marketplace.challenges.tsx (hunk 6 above)
Comment (thehannahlerman) [reply app/routes/app/brainstorm/$marketplace.challenges.tsx:313]: we could add this component to the ProjectBadge component. This I believe is a newer better looking version that Sebastian created for the Brainstorm page

Comment (termaatFlipside) [reply app/routes/app/brainstorm/$marketplace.challenges.tsx]: Oh I didn't realize it was in the table and the top badge. I think the top of page one probably want to be ProjectBadge unless we are changing the design then yeah combine

**Review by bcinman — APPROVED** [review]

**Pull request merged**